### PR TITLE
fix(ui): use noto jp sans font

### LINKED ISSUE #44955: Adding Custom Fonts for Different Langauges
Currently Lato does not support italic for Japanese and we optimally would like to support different fonts for different languages. 

Here is the plan:
1) download the related woff files and place them in the [static directory](https://github.com/freeCodeCamp/freeCodeCamp/tree/main/client/static/fonts).
2) group all font assignments in css into one file for english, and dynamically load alternative files.
3) create [typeface definition css files](https://github.com/freecodecamp/freeCodeCamp/blob/984f311061fd504b5a71310b40d0947fee525a52/client%2Fsrc%2Fcomponents%2Flayouts%2Ffonts.css) for different fonts and load them dynamically.
5) inject [link tags](https://github.com/freecodecamp/freeCodeCamp/blob/4bdfb6794f65b49e6141cef29c8095cd1126c307/client%2Fsrc%2Fcomponents%2Flayouts%2Fdefault.tsx#L159-L200) conditionally based on language. 

Alternatively we could introduce new names to our our fonts, and rename all fonts across css files, and skip number 3.

Feedback is appreciated.

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #44955

My hope is that the Japanese font files actually loaded.  I don't know how to dynamically change the navigator language and I was not aware of any existing tests in this repository that I can leverage for font testing. I will want to add one ideally before this gets merged.

### CODE REVIEW

**Comment by a2937** [comment]
Dang. Well of navigator doesn't exist , I guess there is another way to tell if the page is Japanese.

File: client/src/components/layouts/default.tsx [hunk 1]
```diff
@@ -165,6 +170,12 @@ function DefaultLayout({
 
   const useSystemTheme = fetchState.complete && isSignedIn === false;
 
+  const lightFontUrl =
+    clientLocale == 'japanese' ? jpSansLightURL : latoLightURL;
```
Comment (naomi-lgbt) [inline client/src/components/layouts/default.tsx:174]:
```suggestion
    clientLocale === 'japanese' ? jpSansLightURL : latoLightURL;
```

File: client/src/components/layouts/default.tsx [hunk 2]
```diff
@@ -165,6 +170,12 @@ function DefaultLayout({
...
+  const regularFontUrl =
+    clientLocale == 'japanese' ? jpSansRegularURL : latoRegularURL;
```
Comment (naomi-lgbt) [inline client/src/components/layouts/default.tsx:176]:
```suggestion
    clientLocale === 'japanese' ? jpSansRegularURL : latoRegularURL;
```

File: client/src/components/layouts/default.tsx [hunk 3]
```diff
@@ -165,6 +170,12 @@ function DefaultLayout({
...
+  const boldFontUrl = clientLocale == 'japanese' ? jpSansBoldURL : latoBoldURL;
```
Comment (naomi-lgbt) [inline client/src/components/layouts/default.tsx:177]:
```suggestion
  const boldFontUrl = clientLocale === 'japanese' ? jpSansBoldURL : latoBoldURL;
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
We should rarely, if ever, be using loose equality.

**Review by naomi-lgbt — APPROVED** [review]

**Comment by a2937** [comment]
I still need to write a test for this. But I'm a little stumped on how to write one that will work with what we're doing here; and is globalization friendly.

**Comment by ojeytonwilliams** [comment]
For testing, I think it would make sense to create a unit test and mock env.json. That should let you create the default layout with and without the Japanese locale and confirm that it has the right font links each time.

**Comment by a2937** [comment]
I've hit a road block. The test I wrote doesn't work in the slightest. The font elements are just missing.

**Review by sidemt — CHANGES_REQUESTED** [review]

**Comment by ojeytonwilliams** [comment]
>I've hit a road block. The test I wrote doesn't work in the slightest. The font elements are just missing.

Yeah, this is more involved than I thought. In principle we can setup Mock Service Worker, mock `window.matchMedia`, provide a user in response to the fetch request etc. etc.  It's a lot just to test if a font is used!

While I do think we should improve our client unit tests (and MSW would help with that), it's excessive for this particular feature. We should wait until we have time to improve the tests more generally.

**Review by sidemt — CHANGES_REQUESTED** [review]

**Review by sidemt — CHANGES_REQUESTED** [review]

**Comment by a2937** [comment]
I was afraid of something weird like that happening. I removed the fragment and that seemed to have gotten things working again.

**Review by sidemt — CHANGES_REQUESTED** [review]
Thanks, one last thing.

**Review by sidemt — APPROVED** [review]
LGTM 👍 Thank you so much for your work!

**Pull request merged**